azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -0,0 +1,175 @@
+        finally:
+            PolicyEngineConfigurator._initialized = True
+
+    @staticmethod
+    def _is_policy_supported():
+        distro_info = get_distro()
+        distro_name = distro_info[0]
+        try:
+            distro_version = FlexibleVersion(distro_info[1])
+        except ValueError:
+            raise ValueError
+
+        # Check if the distro is in the support matrix and if the version is supported
+        if distro_name in POLICY_SUPPORT_MATRIX:
+            min_version = POLICY_SUPPORT_MATRIX[distro_name]
+            return distro_version >= min_version
+        else:
+            return False
+
+    @staticmethod
+    def get_instance():
+        if PolicyEngineConfigurator._instance is None:
+            PolicyEngineConfigurator._instance = PolicyEngineConfigurator()
+        return PolicyEngineConfigurator._instance
+
+    @staticmethod
+    def get_policy_enabled():
+        return PolicyEngineConfigurator.get_instance()._policy_enabled
+
+
+class PolicyEngine(object):
+    """
+    Implements policy engine API. Class will always be initialized, but if the Regorus import fails,
+    all methods will be no-ops.
+
+    If any errors are thrown in regorus.py, they will be caught and handled here. add_policy, add_data,
+    and set_input will be no-ops, eval_query will return an empty dict 
+    """
+    def __init__(self):
+        self._policy_engine_enabled = False
+        self._engine = None
+        try:
+            if PolicyEngineConfigurator.get_instance().get_policy_enabled():
+                self._engine = regorus.Engine()  # regorus will have already been imported in configurator
+                self._policy_engine_enabled = True
+        except (ImportError, NameError) as ex:
+            log_policy("Error: Failed to initialize Regorus policy engine due to import failure. {0}".format(ex), is_success=False)
+        except Exception as ex:
+            log_policy("Error: Failed to initialize Regorus policy engine. '{0}'".format(ex), is_success=False)
+
+    @property
+    def policy_engine_enabled(self):
+        return self._policy_engine_enabled
+
+    def add_policy(self, policy_file):
+        """Policy_path should be a path to a valid Rego policy rule file."""
+        if not self.policy_engine_enabled:
+            return
+
+        try:
+            self._engine.add_policy(policy_file)
+        except Exception as ex:
+            log_policy("Error: Failed to add policy to Regorus policy engine. '{0}'".format(ex), is_success=False)
+
+    def add_data(self, data_file):
+        """Data_file should be a path to a valid JSON data file."""
+        if not self.policy_engine_enabled:
+            return
+
+        try:
+            self._engine.add_data(data_file)
+        except Exception as ex:
+            log_policy("Error: Failed to add data to Regorus policy engine. '{0}'".format(ex), is_success=False)
+
+    def set_input(self, input_file):
```

Comment:
create a temp file for set_input and clean it up

Reply:
Added this functionality in regorus.py, set_input in policy_engine.py accepts a dict or json string.